### PR TITLE
18 check syntax

### PR DESCRIPTION
**Pull request by philippschwarzmueller**
it is about time to bring in some of the advanced syntax checking

### CODE REVIEW

File: tests/arg_tests.c
Comment (fsandel) [inline tests/arg_tests.c:1]: would it make sense to apply norm to the test files aswell?

File: src/parser/check_syntax.c [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	while (i < (int) ft_strlen(f_c_str + 2) - 1)
+	{
+		if (!is_valid_color_char(f_c_str[i]))
+			res = false;
+		i++;
+	}
+	return (res);
+}
+
+static bool	is_map_char(char c, int *whitespace)
```
Comment (fsandel) [inline src/parser/check_syntax.c:56]: would it make sense to change it to a 

static bool is_map_char(char c)
return (true/false)

and count the whitespace characters in the parent function? so the function has similar syntax as the other is_... functions?

File: src/parser/parser.c [hunk 2]
```diff
@@ -1,37 +1,46 @@
 #include "cub3D.h"
 
+struct s_parser_state
```
Comment (fsandel) [inline src/parser/parser.c:3]: didnt we want to use typedefs all the time?

File: src/parser/validate_args.c [hunk 3]
```diff
@@ -11,21 +11,24 @@ int	args_valid(int argc, char **argv)
 
 	if (argc != 2)
```
Comment (fsandel) [inline src/parser/validate_args.c:12]: maybe add a message or something here if arg_count is wrong
i know it was imlpemented in an earlier request, but still could be nice :)

File: src/utils/map.c [hunk 4]
```diff
@@ -1,16 +1,25 @@
...
+{
+	map->floor_color = color;
+}
+
+void	set_ceiling_color(t_map *map, int color)
+{
+	map->ceiling_color = color;
+}
+
+void	set_cube_value(t_map *map, int line, int column, char c)
```
Comment (fsandel) [inline src/utils/map.c:13]: replace char c with a enum?

**Review by fsandel — COMMENTED** [review]
Maybe we should talk about the ideas in the comments

File: tests/arg_tests.c
Comment (philippschwarzmueller) [reply tests/arg_tests.c:1]: Tried to, but the tests use some syntax that isn't norm compliant and results in weird norm errors

File: src/parser/parser.c (hunk 2 above)
Comment (philippschwarzmueller) [reply src/parser/parser.c:3]: this state struct is yet to be extended for all the parsing states etc. Up until it is just in use in this file I didn't see a point in making it a typedef in the header file

File: src/utils/map.c (hunk 4 above)
Comment (philippschwarzmueller) [reply src/utils/map.c:13]: the way set cube value is used is that it checks if the char c is a valid cube enum value and if so sets it to the corresponding enum value. this is due to the functions use in the parser - until that behaviour gets us in to trouble somehow i would want to keep it that way

**Pull request merged**